experanto/interpolators.py

```diff
@@ -427,3 +433,86 @@ def __init__(self, file_name, data) -> None:
 
     def get_data(self) -> np.array:
         return np.full((1,) + self.image_size, self.interleave_value)
+
+
+class SpikesInterpolator(Interpolator):
+    def __init__(
+        self,
+        root_folder: str,
+        sampling_rate: float = 100.0,  # Default to 100 Hz
+        normalize: bool = False,
+        **kwargs,
+    ) -> None:
+        """
+        Interpolator specifically designed for spike train data.
+
+        Args:
+            root_folder (str): Path to the root folder containing spike data.
+            sampling_rate (float): Sampling rate in Hz (default: 100 Hz).
+            normalize (bool): Whether to normalize spike counts.
+        """
+        super().__init__(root_folder)
+        meta = self.load_meta()
+        self.sampling_rate = meta.get("sampling_rate", sampling_rate)
+        self.time_delta = 1.0 / self.sampling_rate
+        self.start_time = meta["start_time"]
+        self.end_time = meta["end_time"]
+        self.valid_interval = TimeInterval(self.start_time, self.end_time)
+
+        self.n_units = meta["n_signals"]  # Number of recorded spike units
+        self.normalize = normalize
+
+        # Load spike data from .npy or .mem file
+        if (self.root_folder / "spikes.npy").exists():
+            self._data = np.load(self.root_folder / "spikes.npy")
+        else:
+            self._data = np.memmap(
+                self.root_folder / "data.mem",
+                dtype=meta["dtype"],
+                mode="r",
+                shape=(meta["n_timestamps"], self.n_units),
+            )
+
+        # Load normalization parameters if needed
+        if self.normalize:
+            self._init_normalization()
```

Comment:
again, this is basically same as [lines 89-123](https://github.com/sensorium-competition/experanto/blob/33b3c00ce2307dbeef6b63b33596c1e89512e14b/experanto/interpolators.py#L89-L123) from the SequenceInterpolator

I would rather make a common parent class, inherit both interpolators from it and reuse the function in both interpolators
You basically just need to set `meta["phase_shift_per_signal"] = False` and rename `spikes.npy` -> `data.npy`